libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/RouteComparator.java

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.services.android.navigation.ui.v5;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directions.v5.models.RouteLeg;
+
+import java.util.List;
+
+class RouteComparator {
+
+  private static final int FIRST_ROUTE = 0;
+  private static final int ONE_ROUTE = 1;
+  private final NavigationViewRouter navigationViewRouter;
+
+  RouteComparator(NavigationViewRouter navigationViewRouter) {
+    this.navigationViewRouter = navigationViewRouter;
+  }
+
+  void compare(@NonNull DirectionsResponse response, @Nullable DirectionsRoute chosenRoute) {
+    if (isValidRoute(response)) {
+      List<DirectionsRoute> routes = response.routes();
+      DirectionsRoute bestRoute = routes.get(FIRST_ROUTE);
```

Comment:
You could move the initialization to the call to `obtainMostSimilarRoute`

Reply:
Adjusted the naming in `RouterComparator` to be more clear 👍